lib/IRGen/GenDecl.cpp

```diff
@@ -1448,13 +1448,15 @@ getIRLinkage(const UniversalLinkageInfo &info, SILLinkage linkage,
     return {llvm::GlobalValue::ExternalLinkage, PublicDefinitionVisibility,
             ExportedStorage};
 
+  case SILLinkage::PublicNonABI:
+    return RESULT(WeakODR, Hidden, Default);
```

Comment:
*sigh* okay, apparently, we do not always emit the default thunk as a definition.  When we create it as a declaration, the problem is that we will still give it a WeakODR linkage, which is not a valid declaration linkage.  I guess, that means that we need the conditional return type.